### ISSUE TITLE
proposal: unicode length validators

### ISSUE THREAD

**Issue by day-dreams** [facebook/ent#784]
MinLen() and MaxLen() mean golang string length. For other lauguages(Chinese, Japanese......), it might be more helpful to validate unicode length.  
```golang
// MinLen adds a length validator for this field.
// Operation fails if the length of the string is less than the given value.
func (b *stringBuilder) MinLen(i int) *stringBuilder {
	b.desc.Validators = append(b.desc.Validators, func(v string) error {
		if len(v) < i {
			return errors.New("value is less than the required length")
		}
		return nil
	})
	return b
}
```

For example, when we limit max length of UserName to 8 charactors,  "12345678" and "我的名字有八个字"  are both accepted.
```golang
package main

import (
	"fmt"
	"unicode/utf8"
)

func main() {
	s := "我的名字有八个字"
	fmt.Printf("unicode length:%d, len(string):%d\n", utf8.RuneCountInString(s), len(s)) // unicode length:8, len(string):24
	s = "12345678"
	fmt.Printf("unicode length:%d, len(string):%d\n", utf8.RuneCountInString(s), len(s)) // unicode length:8, len(string):8
}
```

**Comment by a8m** [comment]
Hey @day-dreams and thanks for proposing this!

First, if you're blocked on this, you can add your custom predicate as follows:

```go
func MaxRuneCount(maxLen int) func(s string) error {
	return func(s string) error {
		if utf8.RuneCountInString(s) > maxLen {
			return errors.New("value is more than the max length")
		}
		return nil
	}
}

field.String("name").
	Validate(MaxRuneCount(10)),
field.String("nickname").
	Validate(MaxRuneCount(20)),
```
Then, comes the question, do we really need to add this to the default field validators?

**Comment by roger2hk** [comment]
Please add this custom validator as another example in the documentation if we are not going to add this to the built-in field validators.

**Comment by a8m** [comment]
> Please add this custom validator as another example in the documentation if we are not going to add this to the built-in field validators.

You are welcome to create a PR 😃

**Comment by roger2hk** [comment]
> > Please add this custom validator as another example in the documentation if we are not going to add this to the built-in field validators.
> 
> You are welcome to create a PR 😃

Sure. Created https://github.com/facebook/ent/pull/814.

**Comment by a8m** [comment]
Closed with #814
